### ISSUE TITLE
Dependency / installation issues

### ISSUE THREAD

**Issue by asalt** [JuliaS92/SpatialProteomicsQC#152]
I'm having trouble getting the app working after installing into a fresh virtual environment python 3.10.

Note I did have to `pip install dask[dataframe]` after installation from `pip install .` of this repository.

/SpatialProteomicsQC/py310-env/lib/python3.10/site-packages/param/__init__.py", line 1347, in _validate_class_
    raise ValueError(
ValueError: ClassSelector parameter None value must be an instance of (function, tuple), not <function size at 0x73804b9f3930>.

I did try a few other versions of param as well, but without luck.

```
/SpatialProteomicsQC/py310-env/lib/python3.10/site-packages/param/__init__.py", line 1347, in _validate_class_
    raise ValueError(
ValueError: ClassSelector parameter None value must be an instance of (function, tuple), not <function size at 0x73804b9f3930>.
```

the call to param comes from some portion of loading of hvplot or holoviews I believe.

```
(py310-env)~/t/S/webapp (master)> pip freeze --all
bleach==6.1.0
bokeh==2.4.3
certifi==2024.8.30
charset-normalizer==3.3.2
click==8.1.7
cloudpickle==3.0.0
colorcet==3.1.0
cycler==0.12.1
dask==2024.9.1
dask-expr==1.1.15
datashader==0.16.3
domaps==1.0.1
et-xmlfile==1.1.0
fonttools==4.54.1
fsspec==2024.9.0
holoviews==1.14.9
hvplot==0.10.0
idna==3.10
importlib_metadata==8.5.0
Jinja2==2.11.3
joblib==1.4.2
kiwisolver==1.4.7
llvmlite==0.43.0
locket==1.0.0
Markdown==3.7
MarkupSafe==2.0.1
matplotlib==3.5.3
matplotlib-venn==0.11.6
multipledispatch==1.0.0
natsort==7.1.1
networkx==3.3
numba==0.60.0
numpy==1.26.4
openpyxl==3.1.5
packaging==24.1
pandas==2.2.3
panel==0.14.2
param==1.12.3
partd==1.4.2
patsy==0.5.6
pillow==10.4.0
pip==22.0.2
plotly==4.14.1
pyarrow==17.0.0
pyct==0.5.0
pyparsing==3.1.4
python-dateutil==2.9.0.post0
pytz==2024.2
pyviz_comms==3.0.3
PyYAML==6.0.2
requests==2.32.3
retrying==1.3.4
scikit-learn==1.5.2
scipy==1.14.1
setuptools==59.6.0
six==1.16.0
statsmodels==0.14.1
threadpoolctl==3.5.0
toolz==0.12.1
tornado==6.4.1
tqdm==4.66.5
typing_extensions==4.12.2
tzdata==2024.2
UpSetPlot==0.9.0
urllib3==2.2.3
webencodings==0.5.1
xarray==2023.1.0
zipp==3.20.2
```

**Comment by JuliaS92** [comment]
Hello, yes this is an issue with the holoviews/networksx and an error I have encountered before. Try installing with python 3.8, please. This should work with the specified requirements. If that still does not work, I can post a pip freeze of my own environment.

**Comment by asalt** [comment]
Thank you a fresh venv with python3 .8 worked.

I had tried 3.10 and 3.9 previously (after 3.12 which also failed).

with python 3.8 and `pip install .` of the root of this repository I've successfully loaded and processed an example dataset provided

**Comment by JuliaS92** [comment]
Dependencies are now fixed for python 3.8-3.10
